tests/contrib/pytest_benchmark/test_pytest_benchmark.py

```diff
@@ -0,0 +1,142 @@
+from ddtrace.contrib.pytest_benchmark.constants import STATISTICS_STDDEV
+from ddtrace.contrib.pytest_benchmark.constants import STATISTICS_STDDEV_OUTLIERS
+from ddtrace.contrib.pytest_benchmark.constants import STATISTICS_TOTAL
+from ddtrace.ext.test import TEST_TYPE
+from ddtrace.internal.ci_visibility import CIVisibility
+from tests.ci_visibility.test_encoder import _patch_dummy_writer
+from tests.utils import TracerTestCase
+from tests.utils import override_env
+
+
+class PytestTestCase(TracerTestCase):
+    @pytest.fixture(autouse=True)
+    def fixtures(self, testdir, monkeypatch, git_repo):
+        self.testdir = testdir
+        self.monkeypatch = monkeypatch
+        self.git_repo = git_repo
+
+    def inline_run(self, *args):
+        """Execute test script with test tracer."""
+
+        class CIVisibilityPlugin:
+            @staticmethod
+            def pytest_configure(config):
+                if is_enabled(config):
+                    with _patch_dummy_writer():
+                        assert CIVisibility.enabled
+                        CIVisibility.disable()
+                        CIVisibility.enable(tracer=self.tracer, config=ddtrace.config.pytest)
+
+        with override_env(dict(DD_API_KEY="foobar.baz")):
+            return self.testdir.inline_run(*args, plugins=[CIVisibilityPlugin()])
+
+    def subprocess_run(self, *args):
+        """Execute test script with test tracer."""
+        return self.testdir.runpytest_subprocess(*args)
+
+    def test_span_contains_benchmark(self):
+        """Test with benchmark."""
+        py_file = self.testdir.makepyfile(
+            """
+            import time
+
+            def sum_longer(value):
+                time.sleep(2)
+                return value
+            def test_sum_longer(benchmark):
+                result = benchmark(sum_longer, 5)
+                assert result == 5
+        """
+        )
+        file_name = os.path.basename(py_file.strpath)
+        rec = self.inline_run("--ddtrace", file_name)
+        rec.assertoutcome(passed=1)
+        spans = self.pop_spans()
+
+        assert len(spans) == 4
+        assert spans[0].get_tag(TEST_TYPE) == "benchmark"
+        assert spans[0].get_tag(BENCHMARK_INFO) == "Time"
+        assert spans[0].get_metric(BENCHMARK_MEAN) > 2
+        assert spans[0].get_metric(BENCHMARK_RUN) == 5
+        assert spans[0].get_metric(STATISTICS_HD15IQR) is not None
+        assert spans[0].get_metric(STATISTICS_IQR) is not None
+        assert spans[0].get_metric(STATISTICS_IQR_OUTLIERS) is not None
+        assert spans[0].get_metric(STATISTICS_LD15IQR) is not None
+        assert spans[0].get_metric(STATISTICS_MAX) > 2
+        assert spans[0].get_metric(STATISTICS_MEAN) > 2
+        assert spans[0].get_metric(STATISTICS_MEDIAN) > 2
+        assert spans[0].get_metric(STATISTICS_MIN) > 2
+        assert spans[0].get_metric(STATISTICS_OPS) > 0
+        assert spans[0].get_tag(STATISTICS_OUTLIERS) is not None
```

Comment:
Should this at least check the type of the data, to validate that `set_tag_str()` was called?

Reply:
Good catch! This has been addressed now